mbed_app.json

```diff
@@ -1,15 +1,60 @@
 {
+    "config": {
+        "demo-button": {
+            "macro_name": "DEMO_BUTTON",
+            "required": true
+        },
+        "demo-button-active-low": {
+            "help": "true if the button state is low when pressed, high when released",
+            "macro_name": "DEMO_BUTTON_ACTIVE_LOW",
+            "required": false
+        }
+    },
     "target_overrides": {
         "*": {
             "mcuboot.bootloader-build": 0,
-            "target.mbed_app_start": "0x20000",
-            "target.mbed_app_size": "0xDC000"
+            "target.c_lib": "small",
+            "mcuboot.log-level": "MCUBOOT_LOG_LEVEL_DEBUG",
+            "mbed-trace.enable": true
         },
         "NRF52840_DK": {
-            "mcuboot.max-img-sectors": 256
+            "demo-button": "BUTTON1",
+            "demo-button-active-low": true,
+            "target.mbed_app_start": "0x21000",
+            "target.mbed_app_size": "0xBE000",
+            "mcuboot.primary-slot-address": "0x20000",
+            "mcuboot.slot-size": "0xC0000",
+            "mcuboot.scratch-address": "0xE0000",
+            "mcuboot.scratch-size": "0x20000",
+            "mcuboot.max-img-sectors": "0x180",
+            "mcuboot.read-granularity": 4,
+            "qspif.QSPI_MIN_PROG_SIZE": 4
         },
         "EP_AGORA": {
-            "mcuboot.max-img-sectors": 256
+            "demo-button": "BUTTON1",
+            "demo-button-active-low": true,
+            "target.mbed_app_start": "0x21000",
+            "target.mbed_app_size": "0xBE000",
+            "mcuboot.primary-slot-address": "0x20000",
+            "mcuboot.slot-size": "0xC0000",
+            "mcuboot.scratch-address": "0xE0000",
+            "mcuboot.scratch-size": "0x20000",
+            "mcuboot.max-img-sectors": "0x180",
+            "mcuboot.read-granularity": 4,
+            "qspif.QSPI_MIN_PROG_SIZE": 4
+        },
+        "DISCO_L475VG_IOT01A": {
+            "demo-button": "USER_BUTTON",
+            "demo-button-active-low": true,
+            "target.mbed_app_start": "0x8021000",
+            "target.mbed_app_size": "0xBE000",
```

Comment:
Why is this `0xBE000` and not `0xBF000`? As it stands now, there is a 4kB gap between the end of the main application slot and the start of the scratch space region:

`target.mbed_app_start + target.mbed_app_size = 0x21000 + 0xBE000 = 0xDF000`
`0xE0000 - 0xDF000 = 0x1000`

Is this to allow space for something like KVStore? If not, should we change the default to `0xBF000` in this example to maximize application space?

Reply:
Here is a WIP diagram of the memory map I made for the readme of this demo:

![mbed-mcuboot-demo-memory-map](https://user-images.githubusercontent.com/350500/98159346-67d1ac00-1eaa-11eb-95ab-e51d4a7d3e2c.jpg)

Reply:
Oh, perhaps the 4kB region is for the TLV trailers?

Reply:
Yes I left some space for TLV (4KB should be way more than enough), but didn't precisely calculate its exact size which seems to depend on the configuration (e.g. if swap is enabled, etc.).